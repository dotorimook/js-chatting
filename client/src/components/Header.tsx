import React, { FC } from 'react'
import IComponentProps from 'interfaces/IComponentProps';
import styled from 'styled-components';
import { applyFlexbox } from 'style/mixins';
import theme from 'style/theme';

interface IHeaderProps extends IComponentProps {
  Left?: FC<IComponentProps>;
  title?: string|FC<IComponentProps>;
  Right?: FC<IComponentProps>;
}

const HeaderFrame = styled.div`
  width:100%;
  height:4em;
  border-bottom: 1px solid ${theme.headerBorder};
  position:relative;
  background-color:${theme.header};
  .title-frame,.btn-frame {
    position:absolute;
    width:100%;
    height:100%;
    top:0;
    left:0;
  }
  .title-frame {
    ${applyFlexbox('row','center','center')};
    .title {
      text-align:center;
    }
  }
  .btn-frame {
    ${applyFlexbox('row','space-between','stretch')};
    button {
      width: auto;
      min-width:4em;
    }
  }
`;

const Header:FC<IHeaderProps> = (props:IHeaderProps) => {
  const {Left, Right, title} = props;
  return (
    <HeaderFrame>
      <div className='title-frame'>
        {!!title  &&  <h1 className='title'>{title}</h1>}
      </div>
      <div className='btn-frame'>
        {!! Left   &&  <Left {...props} />}
        <div style={{flex:1}}></div>
        {!! Right &&  <Right {...props} />}
      </div>
    </HeaderFrame>
  )
}

export default Header;