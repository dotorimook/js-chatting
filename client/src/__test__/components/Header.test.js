import React from 'react'
import 'setupTest';
import Header from 'components/Header';
import {shallow} from 'enzyme';

describe('Header Test', () => {
  let ui = null;
  const Left = () => (<div className='left'>leftChildren</div>);
  const Right = () => (<div className='right'>rightChildren</div>);
  const title = 'title';

  beforeAll(()=>{
    ui = shallow(<Header Left={Left} Right={Right} title={title}/>);
  });

  it('matches snapshot', () => {
    expect(ui).toMatchSnapshot();
  });

  it('properly renders title', () => {
    expect(ui.find('.title').length).toBeGreaterThan(0);
    expect(ui.find('.title').text().indexOf(title)).toBeGreaterThanOrEqual(0);
  });

  it('has a left and right', () => {
    expect(ui.find(Left)).toHaveLength(1);
    expect(ui.find(Right)).toHaveLength(1);
  });
})