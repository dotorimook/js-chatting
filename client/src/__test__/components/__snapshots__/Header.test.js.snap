// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test matches snapshot 1`] = `
<styled.div>
  <div
    className="title-frame"
  >
    <h1
      className="title"
    >
      title
    </h1>
  </div>
  <div
    className="btn-frame"
  >
    <Left
      Left={[Function]}
      Right={[Function]}
      title="title"
    />
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <Right
      Left={[Function]}
      Right={[Function]}
      title="title"
    />
  </div>
</styled.div>
`;
